pycode/memilio-epidata/memilio/epidata_test/test_epidata_transform_mobility_data.py

```diff
@@ -0,0 +1,150 @@
+from memilio.epidata import defaultDict as dd
+from memilio.epidata import geoModificationGermany as geoger
+from memilio.epidata import transformMobilityData as tfmd
+
+
+class TestTransformMobilityData(fake_filesystem_unittest.TestCase):
+    maxDiff = None
+
+    path = '/home/Mobility/'
+
+    counties = geoger.get_county_ids(merge_eisenach=False)
+
+    df_401 = pd.DataFrame(np.zeros((len(counties), len(counties))))
+
+    # two neighboring counties (state 0) to one county (state 1)
+    df_401.iloc[0, 15] = 2
+    df_401.iloc[1, 15] = 3
+    # two neighboring counties (state 2) to one county (state 5)
+    df_401.iloc[44, 122] = 5
+    df_401.iloc[45, 122] = 4
+    # one county (state 2) to two neighboring counties (state 11)
+    df_401.iloc[44, 333] = 1
+    df_401.iloc[44, 335] = 7  
+    # Eisenach (state 15) to ... (state 2 and 15)
+    df_401.iloc[383, 27] = 3
+    df_401.iloc[383, 399] = 14
+    # index between Eisenach and Wartburgkreis (state 15) to ... (state 7)
+    df_401.iloc[384, 217] = 22
+    # Wartburgkreis (state 15) to ... (state 2, 15, 2)
+    df_401.iloc[386, 27] = 3
+    df_401.iloc[399, 399] = 27
+    df_401.iloc[399, 24] = 11
+    # state 2 to state 15
+    df_401.iloc[55, 387] = 11    
+
+    def setUp(self):
+        self.setUpPyfakefs()
+    
+
+    # Test that correct data frame is returned by get method
+    @patch('memilio.epidata.transformMobilityData.pd.read_csv',
+           return_value=df_401.copy())
+    def test_get_mobility_from_file(self, mock_load_file):
+        df_return = tfmd.getMobilityFromFile(self.path, mobility_file='mobility')
+        self.assertTrue(df_return.equals(self.df_401))
+
+    # Test that data frame of size 401 is reduced
+    @patch('memilio.epidata.transformMobilityData.getMobilityFromFile',
+           return_value=df_401.copy())
+    def test_update_mobility_reduction_401to400(self, mock_load_file):
+        # read is mocked
+        df_read = tfmd.getMobilityFromFile(self.path, 'mobility')        
+        # create folder where new file can be stored and run updateMobility
+        gd.check_dir(self.path)
+        df_return = tfmd.updateMobility2022(self.path, mobility_file='mobility')
+        # different frame is returned
+        self.assertFalse(df_return.equals(df_read))
+        # reduced frame is returned
+        self.assertEqual(len(df_return), 400)
+        self.assertEqual(len(df_return.columns), 400)
+        # new file is written / file is overwritten
+        self.assertTrue(os.path.isfile(os.path.join(self.path, 'mobility.txt')))
+        # Eisenach is index 383 and Wartburgkreis 386 in old county list
+        self.assertTrue(abs(df_return.iloc[0:383, 0:383].values-df_read.iloc[0:383, 0:383].values).max().max() < 1e-10)
+        self.assertTrue(abs(df_return.iloc[383:385, :383].values-df_read.iloc[384:386, :383].values).max().max() < 1e-10)
+        self.assertTrue(abs(df_return.iloc[386:, 386:].values-df_read.iloc[387:, 387:].values).max().max() < 1e-10)
```

Comment:
@patricklnz Basically you are right. However, we should stick to the auto-formatting that we use. If not, this is all reformatted over and over again. The basic problem is that the auto-formatting has to be improved since it does not work all the time...

Reply:
see #189